Though leaving these features enabled on LLVM17 does not
cause compilation to fail, it floods the log output with
a large number of "not a recognized feature" messages.

Remove this patch when switching to LLVM18.


From 32c68ac95130fc50482fd27f5d75e0d0fc3fb199 Mon Sep 17 00:00:00 2001
From: WANG Rui <wangrui@loongson.cn>
Date: Sun, 28 Apr 2024 15:55:27 +0800
Subject: [PATCH] Revert "loongarch: add frecipe and relax target feature"

This reverts commit e81df3f3225ce53921a87b9e490959c41b783b02.
---
 compiler/rustc_target/src/target_features.rs | 2 --
 tests/ui/abi/compatibility.rs                | 2 +-
 2 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/compiler/rustc_target/src/target_features.rs b/compiler/rustc_target/src/target_features.rs
index c003982278e..af5a565fdc9 100644
--- a/compiler/rustc_target/src/target_features.rs
+++ b/compiler/rustc_target/src/target_features.rs
@@ -377,12 +377,10 @@ pub fn is_stable(self) -> bool {
     // tidy-alphabetical-start
     ("d", Unstable(sym::loongarch_target_feature)),
     ("f", Unstable(sym::loongarch_target_feature)),
-    ("frecipe", Unstable(sym::loongarch_target_feature)),
     ("lasx", Unstable(sym::loongarch_target_feature)),
     ("lbt", Unstable(sym::loongarch_target_feature)),
     ("lsx", Unstable(sym::loongarch_target_feature)),
     ("lvz", Unstable(sym::loongarch_target_feature)),
-    ("relax", Unstable(sym::loongarch_target_feature)),
     ("ual", Unstable(sym::loongarch_target_feature)),
     // tidy-alphabetical-end
 ];
diff --git a/tests/ui/abi/compatibility.rs b/tests/ui/abi/compatibility.rs
index 3ee4542810c..a4f60ea2684 100644
--- a/tests/ui/abi/compatibility.rs
+++ b/tests/ui/abi/compatibility.rs
@@ -39,7 +39,7 @@
 //@ revisions: loongarch64
 //@[loongarch64] compile-flags: --target loongarch64-unknown-linux-gnu
 //@[loongarch64] needs-llvm-components: loongarch
-//@[loongarch64] min-llvm-version: 18
+//@[loongarch64] min-llvm-version: 17
 //@ revisions: wasm
 //@[wasm] compile-flags: --target wasm32-unknown-unknown
 //@[wasm] needs-llvm-components: webassembly
-- 
2.44.0

